### PR TITLE
Discovery: more tests please

### PR DESCRIPTION
**Pull request by pgwillia**
#1103 

With Solr datastore available and ingest tasks run for the test sets, more tests pass. `Coverage report generated for RSpec to ./discovery/coverage. 981 / 2231 LOC (43.97%) covered.`

Depends on #1195

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,22 @@
 language: ruby
+addons:
+  chrome: stable
```
Comment (murny) [inline .travis.yml]: Not sure what the default `sudo` is? 

But if your adding chrome addon to travisci and getting random failure errors, you need to enforce`sudo: required` in your travisci

https://docs.travis-ci.com/user/chrome#selecting-a-chrome-version

More info here:
https://github.com/travis-ci/travis-ci/issues/8836

File: config/blacklight.yml [hunk 2]
```diff
@@ -10,10 +10,15 @@
...
 # how to start up solr, generally for automated testing.
 
-
+default: &default
+  url: <%= Rails.application.secrets.solr_url %>
 development:
-  url: "http://solrcloud-test.library.ualberta.ca:8080/solr/discovery-test"
-test: &test
-  url: <%= Rails.application.config.solr %>
+  url: <%= ENV['SOLR_URL'] || 'http://localhost:8983/solr/discovery' %>
```
Comment (murny) [inline config/blacklight.yml]: Maybe you can consolidate the config here? You have some in ENV vars, some in this file, some in application secrets?

Could just put everything in application secrets?

Than this file just becomes:

```
default: &default
  url: <%= Rails.application.secrets.solr_url %>
development: 
  <<: *default
test: 
  <<: *default
  writable: true
ingest: 
  <<: *default
  writable: true
production: 
  <<: *default
```
etc.

File: spec/controllers/bookmarks_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  describe "delete" do
+    before do
+      @controller.send(:current_or_guest_user).save
+      @controller.send(:current_or_guest_user).bookmarks.create! document_id: '2007020969', document_type: "SolrDocument"
+    end
+    
+    it "has a 200 status code when delete is success" do
+      xhr :delete, :destroy, :id => '2007020969', :format => :js
+      expect(response).to be_success
+      expect(response.code).to eq "200"
```
Comment (murny) [inline spec/controllers/bookmarks_controller_spec.rb]: Could be wrong, but i think these two assertions are exactly the same?
```
expect(response).to be_success
expect(response.code).to eq "200"
```

`be_success` checks if the response was a 200 (compared to a 302 (redirect), 404 (not found) or 5xx (error))


One of the biggest issues with rspec is that you can do the same thing 50 different ways, but only one way is deemed "the right way" by the community.

In RSpec world, i believe the best practice for checking responses is to use `have_http_status` so for example you can use a number code, symbol or generic code (e.g. :error is 5xx)


```
expect(response).to have_http_status(200)
expect(response).to have_http_status(:success)
expect(response).to have_http_status(:ok)
```
More info: https://relishapp.com/rspec/rspec-rails/docs/matchers/have-http-status-matcher

File: spec/controllers/databases_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+require_relative "../rails_helper"
+
+describe DatabasesController, type: :controller do
+
+  it "should have a collection name" do
+    get :index
+    expect(assigns(:collection_name)).to eq "Databases"
```
[the remark this answers is not in the document]
Comment (pgwillia) [reply spec/controllers/databases_controller_spec.rb]: I agree that the tests could be improved a lot.  A bunch of the test suite that I inherited with the project was failing.  My goal with this is that 
* now that the 'test' environment has been reclaimed 
* a Solr docker container with the blacklight_solr_conf is available as a datastore that travis can run
* ingest target is configurable
* tests helper runs a ingest of the sample marc
-> the tests that I had previously removed (#1172) now pass.

**Comment by pgwillia** [comment]
Ingest is getting a bit of a rewrite.  Will incorporate the feedback here in the new PR.

**Pull request closed without merging**